pkg/webhook/mutation/pod/metadata/mutator.go

```diff
@@ -36,7 +38,19 @@ func (mut *Mutator) Enabled(request *dtwebhook.BaseRequest) bool {
 		request.DynaKube.FeatureAutomaticInjection())
 	enabledOnDynakube := request.DynaKube.MetaDataEnrichmentEnabled()
 
-	return enabledOnPod && enabledOnDynakube
+	matchesNamespace := true // if no namespace selector is configured, we just pass set this to true
+
+	if enabledOnDynakube && request.DynaKube.MetadataEnrichmentNamespaceSelector() != nil && request.DynaKube.MetadataEnrichmentNamespaceSelector().Size() > 0 {
```

Comment:
```suggestion
	if request.DynaKube.MetadataEnrichmentNamespaceSelector().Size() > 0 {
```
wouldn't this be enough?
the first line in the `.Size()` implementation has a nil check in it.

you can write a test for that

checking the `enabledOnDynaKube` feels like a bit of optimisation that probably is not worth it, because to me its much cleaner to define each condition `matchesNamespace, enabledOnPod, enabledOnDynakube` one by one, and at the end "combine them" accordingly
- you already check the `enabledOnDynakube` at the end, so doing it here is not fully necessary